app/models/miq_server/worker_management.rb

```diff
@@ -1,19 +1,24 @@
-module MiqServer::WorkerManagement
-  extend ActiveSupport::Concern
+class MiqServer::WorkerManagement
+  include Vmdb::Logging
 
   include_concern 'Dequeue'
   include_concern 'Heartbeat'
   include_concern 'Monitor'
 
-  included do
-    has_many :miq_workers
+  attr_reader :my_server
+
+  def initialize(my_server)
+    @my_server = my_server
   end
 
-  module ClassMethods
-    def kill_all_workers
-      svr = my_server(true)
-      svr.kill_all_workers unless svr.nil?
-    end
+  delegate :id, :guid, :zone, :roles_changed, :active_roles_changed?, :role_changes, :miq_workers, :to => :my_server
```

Comment:
not sure if delegates are good at this point.
I was going to suggest adding `update` but I realized that the delegate removes some pain that maybe you want to feel at this point.

Reply:
Yeah I don't like how many properties I had to delegate up (left this here as a list of stuff I have to deal with, see https://github.com/ManageIQ/manageiq/pull/21445#issuecomment-922048372)

I'm hoping that some of these can move down or I might have to make `WorkerManagement::Monitor` the "pluggable" layer rather than `WorkerManagement`